### ISSUE TITLE
task changes state from KILLED to STARTED

### ISSUE THREAD

**Issue by tgpfeiffer** [retz/retz#69]
I submitted a task and see the following behavior in my logs:

```
2016-11-08 14:24:25.990 INFO  [qtp294049327-27] i.github.retz.scheduler.NaivePlanner - Job 1(task retz-foobar-id-1) is to be ran as '<my-cmd>' at Slave efb5dd1d-0138-4004-9b1f-5c426be0c221-S2
2016-11-08 14:24:25.996 INFO  [Thread-34] i.github.retz.scheduler.NaivePlanner - Job 1(task retz-foobar-id-1) is to be ran as '<my-cmd>' at Slave 9284b207-a55f-4f34-8446-7a0a26bf4e8b-S5
2016-11-08 14:24:25.998 INFO  [Thread-34] io.github.retz.db.Database - TaskId of id=1: retz-foobar-id-1 / retz-foobar-id-1
2016-11-08 14:24:26.007 INFO  [qtp294049327-27] io.github.retz.db.Database - TaskId of id=1: retz-foobar-id-1 / retz-foobar-id-1
2016-11-08 14:24:26.014 INFO  [Thread-34] io.github.retz.db.Database - Job (id=1) status updated to STARTING
2016-11-08 14:24:26.016 INFO  [Thread-34] i.g.retz.scheduler.RetzScheduler - 1 accepted, 0 declined (0 offers back in stock)
2016-11-08 14:24:26.024 INFO  [qtp294049327-27] io.github.retz.db.Database - Job (id=1) status updated to STARTING
2016-11-08 14:24:26.026 INFO  [qtp294049327-27] i.g.retz.scheduler.RetzScheduler - 1 accepted, 0 declined (1 offers back in stock)
2016-11-08 14:24:26.027 INFO  [qtp294049327-27] io.github.retz.web.WebConsole - Job '<my-cmd>' at foobar has been scheduled at 2016-11-08T14:24:25.883+09:00.
2016-11-08 14:24:26.051 INFO  [Thread-35] i.g.retz.scheduler.RetzScheduler - Status update of task retz-foobar-id-1: TASK_ERROR / Task has duplicate ID: retz-foobar-id-1
2016-11-08 14:24:26.091 INFO  [Thread-35] io.github.retz.db.Database - Job (id=1) status updated to KILLED
2016-11-08 14:24:26.092 INFO  [Thread-35] io.github.retz.scheduler.JobQueue - Job id=1 has failed: Task has duplicate ID: retz-foobar-id-1
2016-11-08 14:24:26.103 INFO  [qtp294049327-28] io.github.retz.web.WebConsole - GET /job/1 from 192.168.56.102 Java/1.8.0_91
2016-11-08 14:24:26.418 INFO  [Thread-36] i.g.retz.scheduler.RetzScheduler - Status update of task retz-foobar-id-1: TASK_RUNNING / 
2016-11-08 14:24:26.467 INFO  [Thread-36] io.github.retz.db.Database - Job (id=1) status updated to STARTED
```

I understand that a duplicate ID may kill the job, and actually the client gets an error when submitting that job (which is good), but the job is still submitted to the cluster and executed there without any problems. I guess the status should not change from KILLED to anything else.

**Comment by kuenishi** [comment]
Ah, this looks like a race between Mesos callback thread and Jetty's web service thread - trying to assign same job (with same id) to different offers/slaves. The solution might be either:
- Add an independent thread that is responsible for calculating job assignment, database update and accepting offers
- Synchronize those race condition with mutex-ish primitive or database transacation among threads

Thank you for finding out and reporting!

**Comment by kuenishi** [comment]
Possible scenario:
1. Jetty thread processing `WebConsole::schedule` inserts a job (say, id=1) with `JobQueue.push`
2. `RetzExecutor.resourceOffer` being called with offer B and  fetches jobs with `JobQueue.findFit` getting a job (id=1)
3. Jetty thread enters `scheduler.get().maybeInvokeNow(driver.get(), job);`  and gets a job (id=1)
4. Jetty thread schedules job (id=1) with a stocked offer (say 'offer A'), updating a job as `STARTED`
5. RetzExecutor thread schedules a job (id=1) with an offered offer B updated a job as `STARTED` again.

**Comment by kuenishi** [comment]
Workaround is to set `retz.max.stock = 0` in retz.properties. This prevents any resource stocking (caching), while job latency might be a bit longer as it have to wait for next resource offer.

**Comment by kuenishi** [comment]
e64dafad434aecd7ca is the fix for this.

**Comment by kuenishi** [comment]
For now I'm closing this issue. The fix for this will be included in 0.0.32, which will be released soon. Feel free to reopen this issue if this is not fixed with 0.0.32 (or later).
